gcloud-java-storage/src/test/java/com/google/gcloud/storage/RemoteGcsHelper.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.storage;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.gcloud.AuthCredentials;
+import com.google.gcloud.storage.RemoteGcsHelper.Option.KeyFromClasspath;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Utility to create a remote storage configuration for testing
+ */
+public class RemoteGcsHelper {
+
+  private static final Logger log = Logger.getLogger(RemoteGcsHelper.class.getName());
+  private static final String BUCKET_NAME_PREFIX = "gcloud-test-bucket-temp-";
+  private static final String PROJECT_ID_ENV_VAR = "GCLOUD_TESTS_PROJECT_ID";
+  private static final String PRIVATE_KEY_ENV_VAR = "GCLOUD_TESTS_KEY";
+
+  private final StorageOptions options;
+
+  private RemoteGcsHelper(StorageOptions options) {
+    this.options = options;
+  }
+
+  /**
+   * Returns a {@StorageOptions} object to be used for testing.
+   */
+  public StorageOptions options() {
+    return options;
+  }
+
+  /**
+   * Delete a bucket recursively. Objects in the bucket are listed and deleted until bucket deletion
+   * succeeds or {@code timeout} expires.
+   *
+   * @param storage the storage service to be used to issue requests
+   * @param bucket the bucket to be deleted
+   * @param timeout the maximum time to wait
+   * @param unit the time unit of the timeout argument
+   * @return true if deletion succeeded, false if timeout expired.
+   * @throws InterruptedException if the thread deleting the bucket is interrupted while waiting
+   * @throws ExecutionException if an exception was thrown while deleting bucket or bucket objects
+   */
+  public static Boolean deleteBucketRecursively(Storage storage, String bucket, long timeout,
+      TimeUnit unit) throws InterruptedException, ExecutionException {
+    ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
I tend to avoid checking the time in a loop as it is often cause of troubles. In our case it would have done very little difference. 

The executor's `finalize` method calls `shutdown` and cleans everything. Even though this is most likely being executed as the last instructions of some tests I'll add an explicit `shutdown()` to free resources without relying on GC.